src/main/java/org/broadinstitute/hellbender/Main.java

```diff
@@ -338,6 +338,14 @@ private CommandLineProgram extractCommandLineProgram( final String[] args,
 
         if (args.length < 1 || args[0].equals("-h") || args[0].equals("--help")) {
             printUsage(System.out, classes, commandLineName);
+        } else if ( args.length == 1 && (args[0].equals("-" + SpecialArgumentsCollection.VERSION_FULLNAME) || args[0].equals("--" + SpecialArgumentsCollection.VERSION_FULLNAME))) {
+            new CommandLineProgram(){
+                @Override
+                protected Object doWork() {
+                    System.out.println(String.format("%s v%s", getToolkitName(), getVersion()));
+                    return 0;
+                }
+            }.doWork();
```

Comment:
It would be nice to find a solution that worked for downstream toolkits - I think this would only work if they repackaged GATK classes in their own jar ? I'm not sure I see an obvious solution that doesn't involve a lot more work though.

Either way, what do you think about adding a couple of methods to RuntimeUtils or somewhere that get the toolkit name and version number from the manifest, and then call those from here and from the corresponding CommandLineProgram methods ? It still wouldn't completely solve the downstream toolkit issue, but at least we wouldn't have to spin up a dummy CLP.  Longer term I really want to fix the special argument handling  and delegation in Barclay.

Reply:
BTW, by "from the manifest" I meant "they way they do now", ie., getClass().getPackage().getImplementationVersion().